src/backend/api/Fusion.Resources.Api/Authorization/Requirements/RequestCommentAuthor.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Fusion.Authorization;
+using Fusion.Resources.Domain;
+using Microsoft.AspNetCore.Authorization;
+using System.Threading.Tasks;
+
+namespace Fusion.Resources.Api.Authorization
+{
+    public class RequestCommentAuthor : FusionAuthorizationRequirement, IAuthorizationHandler
+    {
+        public override string Description => "The user needs to be the author of the comment";
+
+        public override string Code => "CommentAuthor";
+
+        public Task HandleAsync(AuthorizationHandlerContext context)
+        {
+            var azureId = context.User.GetAzureUniqueId();
+
+            var resource = context.Resource as QueryRequestComment;
+
+            if (resource?.CreatedBy.AzureUniqueId == azureId)
+                context.Succeed(this);
+
+            SetEvaluation("User is not author of the comment");
```

Comment:
Could maybe detect that the resource is not the expected type, and report that the requirement was not evaluated. 
```
Evaluation = "Invalid comment resource passed in"
WasEvaluated = false
```
?